tests/conftest.py

```diff
@@ -1300,57 +1300,69 @@ def get_inband(self, ctnname):
         return {}
 
     def get_topo_neigh(self):
-        cwd = os.getcwd()
-        neighs_by_ctn = {}
+        instance_to_neighbor_map = {}
         if "neighbor_connections" not in self.virt_topo:
-            return neighs_by_ctn
-        for nck, ncv in self.virt_topo["neighbor_connections"].items():
-            ctndir = nck.split("-")[0]
-            nbrctndir = nck.split("-")[1]
-            nbrvethintf = int(ncv[nbrctndir].split("eth")[1])
-            nbrintf = "ethernet%d|" % ((nbrvethintf - 1) * 4)
-            vethintf = int(ncv[ctndir].split("eth")[1])
-            cfgdir = cwd + "/virtual_chassis/" + ctndir
-            cfgfile = cfgdir + "/default_config.json"
-            ctnname = ""
-            with open(cfgfile, "r") as cfg:
-                defcfg = json.load(cfg)["DEVICE_METADATA"]["localhost"]
-                ctnname = defcfg["hostname"] + "." + self.ns
-            cfgfile = cwd + "/virtual_chassis/" + nbrctndir + "/default_config.json"
-            with open(cfgfile, "r") as cfg:
-                intfCfg = json.load(cfg)["INTERFACE"]
-                for key in intfCfg:
-                    nbraddr = ""
-                    if key.lower().startswith(nbrintf):
-                        intfaddr = re.split("/|\\|", key)
-                        if len(intfaddr) > 1:
-                            nbraddr = intfaddr[1]
-                        if nbraddr == "":
+            return instance_to_neighbor_map
+
+        working_dir = os.getcwd()
+        for conn, endpoints in self.virt_topo["neighbor_connections"].items():
+            chassis_instance = conn.split('-')[0]
+            neighbor_instance = conn.split('-')[1]
+
+            chassis_config_dir = os.path.join(working_dir, "virtual_chassis", chassis_instance)
+            chassis_config_file = os.path.join(chassis_config_dir, "default_config.json")
+            instance_container_name = ""
+            with open(chassis_config_file, "r") as cfg:
+                device_info = json.load(cfg)["DEVICE_METADATA"]["localhost"]
+                instance_container_name = device_info["hostname"] + "." + self.ns
+
+            chassis_veth_intf = int(endpoints[chassis_instance].split("eth")[1])
+            chassis_host_intf = f"Ethernet{(chassis_veth_intf - 1) * 4}"
+
+            chassis_config_file = os.path.join(working_dir, "virtual_chassis", neighbor_instance, "default_config.json")
```

Comment:
the variable needs to be neighbor_config_file

Reply:
Doesn't this config file reside in the LC instance, not the neighbor?

Reply:
Ah, I see now. Thanks!

Reply:
each instance has its own config file. so the chassis_instance and neighbor_instance configs are separate

Reply:
Fixed